actor-tests/src/test/scala/org/apache/pekko/actor/ActorSystemSpec.scala

```diff
@@ -135,6 +135,7 @@ class ActorSystemSpec extends PekkoSpec(ActorSystemSpec.config) with ImplicitSen
       for (n <- Seq(
           "-hallowelt",
           "_hallowelt",
+          "hallo welt",
           "hallo*welt",
           "hallo@welt",
           "hallo#welt",
```

Comment:
This maybe better with some random generation  properties tests.

Reply:
You mean to replace all those strings with a random generation?

Reply:
Yes, I think that should be better, what do you think.

Reply:
```scala
  if (!name.matches("""^[a-zA-Z0-9][a-zA-Z0-9-_]*$"""))
    throw new IllegalArgumentException(
      "invalid ActorSystem name [" + name +
      "], must contain only word characters (i.e. [a-zA-Z0-9] plus non-leading '-' or '_')")
```

Reply:
Yes, agree